activerecord/lib/active_record/locking/pessimistic.rb

```diff
@@ -9,6 +9,8 @@ module Locking
     # lock on the selected rows:
     #   # select * from accounts where id=1 for update
     #   Account.lock.find(1)
+    #   # select * from accounts where name="grace" for update
+    #   Account.lock.where(name: "grace").load
```

Comment:
Can you remove the `load`. We don't use it in the documentation.

Reply:
removing `load` makes the query not be executed, maybe `to_a`?

Reply:
We are not talking about when the query is executed or not, we are just showing what would be the generated query when it is executed. `to_a` or `load` is just noise here.

Reply:
so, that's why I did first in my code expecting the query to get executed (like `.delete_all` for instance) so putting it in the doc without some kind of loading would be confusing to users I think.

maybe an example with `find_each` would be better?

Reply:
Hmmm, good point about it being confusing, but I still don't think it is in the responsibility of the `lock` documentation to teach users about the lazy evaluation nature of Active Record. I believe this is covered in the getting started guide, it is not?

Reply:
maybe a `lock!` method would be more appropriate as `lock` is just a scope basically (`lock!` exists on active record instances but not as class methods).

Reply:
ah, I think now I got your point. Because `lock` is lazy evaluated your code was expecting the `lock` call to actually acquire the lock but it only did some time later when you executed the query, and that was already too late. Was that the case?

Reply:
I expected `Model.something.lock` to lock my model, method should be named `Model.something.with_lock` to not imply a direct action.

Reply:
`with_lock` also exists. The direct action of `lock` is there, it is just lazy. `Model.something.lock` will lock you model as soon the query is executed.

Reply:
```
>> User.with_lock
Traceback (most recent call last):
        2: from (irb):3
        1: from (irb):3:in `rescue in irb_binding'
NoMethodError (undefined method `with_lock' for #<Class:0x00007fbae053eda8>)
```

`with_lock` is defined on the instances of the models, not on the classes

https://edgeapi.rubyonrails.org/classes/ActiveRecord/Locking/Pessimistic.html